frontend/src/components/Contexts/MissionFilterContext.tsx

```diff
@@ -41,117 +60,261 @@ const completedStatuses = [
+                        ' ' +
+                        TranslateText('minStartTime')
+                )
+            else {
+                setFilterIsSet(true)
+                setFilterState({ ...filterState, maxStartTime: newMaxStartTime })
+            }
+        },
+        switchMinEndTime: (newMinEndTime: number | undefined) => {
+            if (filterState.maxEndTime && newMinEndTime && newMinEndTime > filterState.maxEndTime)
+                setFilterError(
+                    TranslateText('minEndTime') +
+                        ' ' +
+                        TranslateText('cannot be greater than') +
+                        ' ' +
+                        TranslateText('maxEndTime')
+                )
+            if (filterState.minStartTime && newMinEndTime && newMinEndTime < filterState.minStartTime)
+                setFilterError(
+                    TranslateText('minEndTime') +
+                        ' ' +
+                        TranslateText('cannot be less than') +
+                        ' ' +
+                        TranslateText('minStartTime')
+                )
+            else {
+                setFilterIsSet(true)
+                setFilterState({ ...filterState, minEndTime: newMinEndTime })
+            }
+        },
+        switchMaxEndTime: (newMaxEndTime: number | undefined) => {
+            if (
+                filterState.maxStartTime &&
+                newMaxEndTime &&
+                filterState.minEndTime &&
+                (newMaxEndTime < filterState.maxStartTime || newMaxEndTime < filterState.minEndTime)
+            )
+                setFilterError(
+                    TranslateText('maxEndTime') +
+                        ' ' +
+                        TranslateText('cannot be less than') +
+                        ' ' +
+                        TranslateText('minEndTime') +
+                        ' ' +
+                        TranslateText('or') +
+                        TranslateText('minEndTime')
+                )
+            else {
+                setFilterIsSet(true)
+                setFilterState({ ...filterState, maxEndTime: newMaxEndTime })
+            }
+        },
+        resetFilters: () => {
+            setFilterIsSet(false)
+            setFilterState(defaultMissionFilterInterface.filterState)
+        },
+        resetFilter(filterName: string) {
+            filterName = filterName.trim()
+            const defaultState = defaultMissionFilterInterface.filterState
+            const defaultValue = defaultState[filterName as keyof typeof defaultState]
+            setFilterState({ ...filterState, [filterName]: defaultValue })
+        },
+        removeFilter(filterName: string) {
+            filterName = filterName.trim()
+            const defaultState = defaultMissionFilterInterface.filterState
+            const defaultValue = defaultState[filterName as keyof typeof defaultState]
+            const newValue = Array.isArray(defaultValue) ? [] : undefined
```

Comment:
Why are we using newValue, isn't it better to just use defaultValue? If we remove the filter on mission status I would prefer going back to default value

Reply:
I remove the element, since we alternatively have resetFilter which resets it to default